ergo-installer.sh

```diff
@@ -0,0 +1,200 @@
+                exit 1
+            fi
+            ;;
+        --api-key=?*)
+            API_KEY=${1#*=} # Delete everything up to "=" and assign the remainder.
+            ;;
+        --api-key=)         # Handle the case of an empty --api-key=
+            echo 'ERROR: "--api-key" requires a non-empty option argument.'
+            exit 1
+            ;;
+
+        --app-dir)
+            if [ "$2" ]; then
+                APP_DIR=$2
+                shift
+            else
+                echo 'ERROR: "--app-dir" requires a non-empty option argument.'
+                exit 1
+            fi
+            ;;
+        --app-dir=?*)
+            NODE_NAME=${1#*=} # Delete everything up to "=" and assign the remainder.
+            ;;
+        --app-dir=)         # Handle the case of an empty --app-dir=
+            echo 'ERROR: "--app-dir" requires a non-empty option argument.'
+            exit 1
+            ;;
+
+        --node-name)
+            if [ "$2" ]; then
+                NODE_NAME=$2
+                shift
+            else
+                echo 'ERROR: "--node-name" requires a non-empty option argument.'
+                exit 1
+            fi
+            ;;
+        --node-name=?*)
+            NODE_NAME=${1#*=} # Delete everything up to "=" and assign the remainder.
+            ;;
+        --node-name=)         # Handle the case of an empty --node-name=
+            echo 'ERROR: "--node-name" requires a non-empty option argument.'
+            exit 1
+            ;;
+
+        --mode)
+            if [ "$2" ]; then
+                MODE=$2
+                shift
+            else
+                echo 'ERROR: "--mode" requires a non-empty option argument.'
+                exit 1
+            fi
+            ;;
+        --mode=?*)
+            MODE=${1#*=} # Delete everything up to "=" and assign the remainder.
+            ;;
+        --mode=)         # Handle the case of an empty --mode=
+            echo 'ERROR: "--mode" requires a non-empty option argument.'
+            exit 1
+            ;;
+
+        --tor)
+            TOR=yes
+            ;;
+
+        --)              # End of all options.
+            shift
+            break
+            ;;
+        -?*)
+            printf 'WARN: Unknown option (ignored): %s\n' "$1" >&2
+            ;;
+        *)               # Default case: No more options, so break out of the loop.
+            break
+    esac
+
+    shift
+done
+
+if [ -z "${API_KEY}" ]; then
+  echo 'ERROR: "--api-key" is required and requires a non-empty option argument.'
+  exit 1
+fi
+
+if [ "${MODE}" != 'full' ] && [ "${MODE}" != 'mining' ]; then
+  echo 'ERROR: "--mode" requires a non-empty option argument that can be "full" or "mining" only.'
+  exit 1
+fi
+
+echo "Ergo node with config file will be installed into '${APP_DIR}' directory and will be named as '${NODE_NAME}' and has API key '${API_KEY}'."
+
+
+## Check preinstalled software
+INSTALLATION_RECOMMENDATIONS=
```

Comment:
@andyceo add additional check of operating system and fail fast when it is not supported by the script.

Reply:
It is not possible as I can say for now... Script designed to be runnable on wide range of OSes. So if OS has shell (even Windows 10 has, but I not tested script on W10) and basic popular GNU software installed, script will download and run Ergo node.

Installation recommendations done for Debian-based systems, as I only sure how to install related software on Ubuntu/Debian, and that was done only for example. It is possible to add more recommendations for MacOS or Windows or other popular OSes, PRs are welcome!

So I coclude - if script run, than it executes as expected on almost every OS.